server/etcdserver/api/etcdhttp/health.go

```diff
@@ -193,3 +224,135 @@ func checkAPI(lg *zap.Logger, srv ServerHealth, serializable bool) Health {
 	lg.Debug("serving /health true")
 	return h
 }
+
+type HealthCheck func(ctx context.Context) error
```

Comment:
High level note on the code. No need to address it now.

Code below has a mixed order, sometimes the calling function is below (installHttpEndpoint after runHealthChecks) and sometimes above (InstallHttpEndpoints before installHttpEndpoint).

Recommend use a single style. Either top to bottom or bottom to top. My personal preference is top to bottom, because we no longer are bounded by C language limits. It helps putting the top most important function on the top of the file, while the implementation details and helpers on the bottom.

Reply:
ack